src/main/java/org/radarcns/stream/KStreamWorker.java

```diff
@@ -126,8 +127,12 @@ protected Properties getStreamProperties(@Nonnull StreamDefinition definition) {
             localClientId += '-' + window.sizeMs + '-' + window.advanceMs;
         }
 
-        return kafkaProperty.getStreamProperties(localClientId, numThreads,
+        Properties props = kafkaProperty.getStreamProperties(localClientId, numThreads,
                 DeviceTimestampExtractor.class);
+        props.put(StreamsConfig.COMMIT_INTERVAL_MS_CONFIG,
+                String.valueOf(definition.getCommitIntervalMs()));
```

Comment:
Perhaps add a random interval proportional to the timeout (like `U([0.75, 1.25]) * interval`), to avoid all commits from all streams triggering at once.

Reply:
good idea